src/common/OracleLib.sol

```diff
@@ -20,4 +20,26 @@ library OracleLib {
         if (NumberUtils.mulDivCheckOverflow(paymentTokenQuantity, decimalMult, assetTokenQuantity)) return 0;
         return mulDiv(paymentTokenQuantity, decimalMult, assetTokenQuantity);
     }
+
+    function applyPriceAssetToPayment(uint256 assetTokenQuantity, uint256 price, uint8 paymentTokenDecimals)
+        internal
+        pure
+        returns (uint256)
+    {
+        // Adjusts the payment token decimals to match the asset token decimals (18) and removes 18 decimals from the result.
+        uint256 decimalMult = 10 ** (36 - paymentTokenDecimals);
```

Comment:
I'd like to understand if we specifically chose the 36 approach for code size optimization? If not, we could make the decimal handling more explicit by separating the adjustments

Reply:
This math is a consequence of `price` having 18 decimals

Reply:
And asset token also having 18 decimals